src/services/router.ts

```diff
@@ -0,0 +1,109 @@
+import { useHistory } from 'react-router-dom';
+import { Token } from 'services/observables/tokens';
+import { wethToken, ethToken } from 'services/web3/config';
+
+const liquidityBase = '/pools/add-liquidity/';
+const rewardsBase = '/portfolio/rewards/';
+export const swap = '/swap';
+export const tokens = '/tokens';
+export const pools = '/pools';
+export const vote = '/vote';
+export const fiat = '/fiat';
+export const tos = '/terms-of-use';
+export const privacyPolicy = '/privacy-policy';
+export const portfolio = '/portfolio';
+export const addLiquidity = `${liquidityBase}:id`;
+export const addLiquidityError = `${liquidityBase}error`;
+export const portfolioRewardsClaim = `${rewardsBase}claim`;
+export const portfolioRewardsStake = `${rewardsBase}stake/:id`;
+
+export const addLiquidityByID = (id: string) => `${liquidityBase}${id}`;
+export const portfolioRewardsStakeByID = (id: string) =>
+  `${rewardsBase}stake/${id}`;
+export const portfolioRewardsStakeByIDnPos = (id: string, pos: string) =>
+  `${rewardsBase}${id}?posGroupId=${pos}`;
+
+export const useNavigation = () => {
+  const history = useHistory();
+
+  const push = (url: string) => {
+    if (url !== window.location.search) history.push(url);
+  };
+
+  const pushPortfolio = () => push(portfolio);
```

Comment:
why did you prefer the push semantic? We abstract away the history stack. I'd expect "goTo" or "navigateTo"